lib/ansible/modules/apt_repository.py

```diff
@@ -296,14 +296,12 @@ def _parse(self, line, raise_if_invalid_or_disabled=False):
         # Check for another "#" in the line and treat a part after it as a comment.
         i = line.find('#')
         if i > 0:
-            comment = line[i + 1:].strip()
-            line = line[:i]
+            line, comment = re.split('#', line, maxsplit=1)
 
-        # Split a source into substring to make sure that it is source spec.
-        # Duplicated whitespaces in a valid source spec will be removed.
-        source = line.strip()
+        # Check if a line is a source spec and if it is, mark it as a valid source line.
+        source = line
         if source:
-            chunks = source.split()
+            chunks = re.split(' ', source, maxsplit=1)
```

Comment:
I am replacing the `_parse` function altogether in https://github.com/ansible/ansible/pull/85881/ so this change is not required.